geps/gep-1651.md

```diff
@@ -0,0 +1,408 @@
+are created. This can be overriden by setting the Gateway's `spec.infrastructure.routability`.
+
+#### Unsupported routability & address values
+
+If a Gateway is unable to provide an address for the desired routability it MUST set the condition `Accepted`
+to `False` with `Reason` set to `UnsupportedRoutability`
+
+#### Status.Addresses
+
+If a Gateway supports the desired 'routability' implementations MUST populate the `status.addresses` with
+an address that has the same routable semantics. The GatewayAddress field `routability` MUST be populated.
+
+Implementations MAY add additional addresses in status, but they MUST be semantically less than the scope
+of the requested scope. For example if a user requests a `Cluster` routable Gateway then the list of addresses
+MUST NOT have a routability of `Public` or `Private`.
+
+We plan on introducing a new type `GatewayStatusAddress` and change Gateway's `status.addresses` to be
+`[]GatewayStatusAddress`. This will allow the status address type to evolve separately from the spec address.
+
+#### In-cluster Gateways and 'Private' Routability
+
+It is RECOMMENDED that in-cluster gateways SHOULD NOT support 'Private' routability. Kubernetes doesn't have
+a concept of 'Private' routability for Services. In the future this may change upstream.
+
+## Examples
+
+#### 1. Request a GatewayAddress that is routable within the same cluster
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    routability: Cluster
+  listeners:
+  - protocol: HTTP
+    port: 80
+```
+
+#### 2. Request a GatewayAddress with a specific routability and address
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    routability: Cluster
+  listeners:
+  - protocol: HTTP
+    port: 80
+  addresses:
+  - value: 10.0.0.8
+```
+
+#### 3. Request a GatewayAddress that is routable on the public internet
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    routability: Public
+  listeners:
+  - protocol: HTTP
+    port: 80
+```
+
+#### 4. Request a GatewayAddress that is a cloud provider's VPC
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    routability: Private
+  listeners:
+  - protocol: HTTP
+    port: 80
+```
+
+## Alternatives
+
+### Introducing new GatewayAddress Types
+
+We could introduce additional `AddressTypes` (ie. `ClusterLocalIPAddress`) but
+this would lead to a combinatorial explosion as new dimensions (ie. IPv6) are
+introduced.
+
+From: https://github.com/kubernetes-sigs/gateway-api/pull/1653#issuecomment-1451246877
```

Comment:
Please make this a link since that doesn't happen automatically with our docs gen (https://deploy-preview-1653--kubernetes-sigs-gateway-api.netlify.app/geps/gep-1651/#introducing-new-gatewayaddress-types)